### PR TITLE
Automated testing after build with Autohat project.

### PR DESCRIPTION
**Pull request by lifeeth**
Connects to #9 

This tests the image generated with AutoHat project by running the qemux86-64.robot test cases. A failure in the test cases will cause the build to fail.

### CODE REVIEW

File: tests/start.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+cd autohat
+MACHINE=${JOB_NAME#yocto-}
```
Comment (agherzan) [inline tests/start.sh]: I would use here device type to not confuse it with yocto machine. Even though they should be the same, there are exceptions (ex rpi).

File: tests/start.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+if [ "$MACHINE" == "qemux86-64" ]; then
+	docker build -t autohat_$MACHINE\_$BUILD_NUMBER .
+	docker run -it --rm -v <path_to_repo>:/autohat --privileged \
+		--env RESINRC_RESIN_URL=${RESINRC_RESIN_URL} \
+		--env email=${RESIN_EMAIL} \
+		--env password=${RESIN_PASSWORD} \
+		--env device_type=${MACHINE} \
+		--env application_name=${MACHINE//-} \
+		--env image=/autohat/resin.img \
+		autohat robot --exitonerror /autohat/example.robot
```
Comment (agherzan) [inline tests/start.sh]: Why do we run an example? Don't we have actual tests?

File: tests/start.sh (hunk 1 above)
Comment (lifeeth) [inline tests/start.sh]: Makes sense! Will do.

Comment (lifeeth) [inline tests/start.sh]: Wait - one problem is that the caller also does the MACHINE in the same way.

File: tests/start.sh [hunk 3]
```diff
@@ -8,16 +8,21 @@ MACHINE=${JOB_NAME#yocto-}
...
 		--env password=${RESIN_PASSWORD} \
 		--env device_type=${MACHINE} \
 		--env application_name=${MACHINE//-} \
 		--env image=/autohat/resin.img \
-		autohat robot --exitonerror /autohat/example.robot
-	docker rmi autohat_$MACHINE\_$BUILD_NUMBER
+		$AUTOHAT_IMAGE robot --exitonerror /autohat/example.robot
 fi
+
+trap cleanup EXIT
```
Comment (agherzan) [inline tests/start.sh]: What if the build is interrupted manually? Can we catch these signals too?

File: tests/start.sh (hunk 1 above)
Comment (agherzan) [inline tests/start.sh]: Don't understand. Who exactly?

File: tests/start.sh (hunk 3 above)
Comment (lifeeth) [inline tests/start.sh]: @agherzan please look at the over all change - I removed this with signals later on :) - Commit by commit is bad in this PR.

**Comment by agherzan** [comment]
LGTM

**Pull request merged**